content/en/docs/tutorials/configuration/updating-configuration-via-a-configmap.md

```diff
@@ -0,0 +1,514 @@
+```
+
+Edit the ConfigMap:
+```shell
+kubectl edit configmap fruits
+```
+
+In the editor that appears, change the value of key `fruits` from `apples` to `mangoes`. Save your changes.
+The kubectl tool updates the ConfigMap accordingly (if you see an error, try again).
+
+Here's an example of how that manifest could look after you edit it:
+```yaml
+apiVersion: v1
+data:
+  fruits: mangoes
+kind: ConfigMap
+# You can leave the existing metadata as they are.
+# The values you'll see won't exactly match these.
+metadata:
+  creationTimestamp: "2024-01-04T16:04:19Z"
+  name: fruits
+  namespace: default
+  resourceVersion: "1749472"
+
+```
+
+You should see the following output:
+```
+configmap/fruits edited
+```
+
+Tail the logs of the Deployment and observe the output for few seconds:
+```shell
+kubectl logs deployment/configmap-env-var
+```
+
+Notice that the output remains **unchanged**, even though you edited the ConfigMap:
+```
+Thu Jan  4 16:12:56 UTC 2024 The basket is full of apples
+Thu Jan  4 16:13:06 UTC 2024 The basket is full of apples
+Thu Jan  4 16:13:16 UTC 2024 The basket is full of apples
+Thu Jan  4 16:13:26 UTC 2024 The basket is full of apples
+
+```
+
+{{< note >}}
+Although the value of the key inside the ConfigMap has changed, the environment variable in the Pod still shows the earlier value. This is because environment variables for a process running inside a Pod are **not** updated when the source data changes; if you wanted to force an update, you would need to have Kubernetes replace your existing Pods. The new Pods would then run with the updated information.
+{{< /note >}}
+
+You can trigger that replacement. Perform a rollout for the Deployment, using
+[`kubectl rollout`](/docs/reference/kubectl/generated/kubectl_rollout/):
+```shell
+# Trigger the rollout
+kubectl rollout restart deployment configmap-env-var
+
+# Wait for the rollout to complete
+kubectl rollout status deployment configmap-env-var --watch=true
+```
+
+Next, check the Deployment:
+```shell
+kubectl get deployment configmap-env-var
+```
+
+You should see an output similar to:
+```
+NAME                READY   UP-TO-DATE   AVAILABLE   AGE
+configmap-env-var   3/3     3            3           12m
+```
+
+Check the Pods:
+```shell
+kubectl get pods --selector=app.kubernetes.io/name=configmap-env-var
+```
+
+The rollout caused Kubernetes to make a new {{< glossary_tooltip term_id="replica-set" text="ReplicaSet" >}} for the Deployment; that means that the existing Pods were terminated, and new ones have been created. You should finally see an output similar to:
+```
+NAME                                 READY   STATUS        RESTARTS   AGE
+configmap-env-var-6d94d89bf5-2ph2l   1/1     Running       0          13s
+configmap-env-var-6d94d89bf5-74twx   1/1     Running       0          8s
+configmap-env-var-6d94d89bf5-d5vx8   1/1     Running       0          11s
+```
+
+View the logs for a Pod in this Deployment:
```

Comment:
There's a problem: we haven't waited for the previous Pods to terminate.

This works, but it's a bit long to type:
```shell
kubectl logs --follow \
  "$( kubectl get pods --selector=app.kubernetes.io/name=configmap-env-var \
  --template '{{range .items}}{{ if not .metadata.deletionTimestamp }}pods/{{.metadata.name}}{{"\n"}}{{end}}{{end}}' |
  head -n 1 )"
```

Reply:
Thanks for the alternate command, but I agree it is too much to type for this tutorial and we would not want to confuse the readers.

I will handle it by modifying the text to clearly imply that the readers should wait till the older Pods terminate completely before proceeding with the tutorial.

Reply:
I've marked this unresolved until a fix is confirmed.

Reply:
I've addressed this in the latest commit.